app/DAO/BakeryDatabase.scala

```diff
@@ -89,60 +90,76 @@ class BakeryDatabase @Inject() (
       .unsafeToFuture()
   }
 
-  def createProduct(newProduct: JsValue): Int = {
+  def createProduct(newProduct: productUpdateRequest): Int = {
 
     /** Steps for creating a record
       *  1. bring in the json, everything should be provided except for the uuid
       *  2. use query.update to commit the record
       *  3. run the transaction
       *  4. return a variable the controller can check to know the transaction succeeded (201 Created)
       */
-    val newName = (newProduct \ "name").asOpt[String]
-    val newQty = (newProduct \ "quantity").asOpt[Int]
-    val newPrice = (newProduct \ "price").asOpt[Double]
+    val newName = newProduct.name
+    val newQty = newProduct.quantity
+    val newPrice = newProduct.price
     val newStamp = OffsetDateTime.now()
-    sql"""INSERT INTO product VALUES (gen_random_uuid(), $newName, $newQty, $newPrice, $newStamp, $newStamp)""".update.run
-      .transact(bakeryTransactor.xa)
-      .unsafeRunSync()
+    (newName, newQty, newPrice) match {
+      case (Some(name), Some(quantity), Some(price)) =>
+        sql"""INSERT INTO product VALUES (gen_random_uuid(), $name, $quantity, $price, $newStamp, $newStamp)""".update.run
+          .transact(bakeryTransactor.xa)
+          .unsafeRunSync()
+      case _ => 0
```

Comment:
Good enough for now, but in the future your users will need more guidance. How do you differentiate between something going wrong in the transaction vs not having the correct items specified (only have 2 for example). A simple solution could be by specifying the error message in the controller with a "please try again, make sure to have x, y and z present in your request"